### PR TITLE
Update CI tests to run against OpenShift cluster

### PR DESCRIPTION
**Pull request by spadgett**

### CODE REVIEW

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
We almost have the crud tests working against our OpenShift cluster. (I had to comment out one special route.)

OLM tests are failing. @alecmerdler Seen this before? I hit this when I run locally, too.

```
1) Interacting with the etcd OCS : creates etcd Operator `Deployment`
   Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
       at ontimeout timers.js:427:11
       at tryOnTimeout timers.js:289:5
       at listOnTimeout timers.js:252:5
       at Timer.processTimers timers.js:212:10
   Error: Timeout - Async callback was not invoked within timeout specified by 
```

@nicolethoen FYI

**Comment by spadgett** [comment]
The clusterroles page renders, but there is this JS error in the logs, which seems to cause the failure.

```
Level { name_: 'SEVERE', value_: 1000 } 'http://localhost:9000/path-J5s26bmCAR/k8s/cluster/clusterroles/view 0:0 Uncaught DOMException: Failed to get ServiceWorkerRegistration objects: No URL is associated with the caller\'s document.'
```

I can't reproduce locally.

**Comment by spadgett** [comment]
GUI-Tests passing cleanly now with nothing commented out.

File: frontend/public/components/app.jsx [hunk 1]
```diff
@@ -180,7 +180,6 @@ class App extends React.PureComponent {
 
           <Route path="/k8s/cluster/clusterroles/:name/add-rule" exact component={EditRulePage} />
           <Route path="/k8s/cluster/clusterroles/:name/:rule/edit" exact component={EditRulePage} />
-          <Route path="/k8s/cluster/clusterroles/:name" component={props => <ResourceDetailsPage {...props} plural="clusterroles" />} />
```
Comment (spadgett) [inline frontend/public/components/app.jsx]: This is unnecessary. The generic rule below handles this (along with other cluster scoped resources) in exactly the same way:

```jsx
<Route path="/k8s/cluster/:plural/:name" component={ResourceDetailsPage} />
```

https://github.com/openshift/console/blob/master/frontend/public/components/app.jsx#L202

Comment (spadgett) [reply frontend/public/components/app.jsx]: Having said that, I can't explain why this fixes the service worker error, which I don't like

**Comment by spadgett** [comment]
@alecmerdler PTAL. OLM tests are not passing, but they aren't against teamui either. This gets us closer.

File: test-gui.sh [hunk 2]
```diff
@@ -2,14 +2,17 @@
 
 set -euo pipefail
 
-source ./contrib/environment.sh
```
Comment (spadgett) [inline test-gui.sh]: The environment.sh script was setting env vars that don't work for OpenShift clusters.

Comment (alecmerdler) [reply test-gui.sh]: We have `oc-environment.sh`, that's what I use.

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
Ugh, hit the service worker error again even though the previous test ran cleanly

**Comment by spadgett** [comment]
jenkins rebuild

**Pull request merged**